### PR TITLE
Update state to treat /profile as a independant section

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Open the env-switcher and click 'More', it should open the profile view.
Close the profile view.
Switch to the machine view
Open the env-switcher and click 'More'. It should open the profile view.
Click the shopping cart.
It should close the profile view.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3989/
Test FAILed.

**Comment by hatched** [comment]
cb() not called :(

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3993/
Test PASSed.

File: jujugui/static/gui/src/app/views/state.js [hunk 1]
```diff
@@ -671,7 +671,11 @@ YUI.add('juju-app-state', function(Y) {
           indexes.push(idx);
         }
       });
-      indexes.sort();
+      // JavaScript array sort sorts alphabetically which causes issues if
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/state.js]: sorts sorts sorts sorts sorts sorts sorts sorts sorts

**Comment by huwshimi** [comment]
QA OK :+1: :shipit:!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**